install.sh

```diff
@@ -1,34 +1,49 @@
 #!/bin/sh
 set -e
-if [ "$1" != "" ]; then
-	if [ -d "$1" ]; then
-		mkdir -p $1/tools/agbcc
-		mkdir -p $1/tools/agbcc/bin
-		mkdir -p $1/tools/agbcc/include
-		mkdir -p $1/tools/agbcc/lib
-		cp agbcc $1/tools/agbcc/bin/
-		cp old_agbcc $1/tools/agbcc/bin/
-		cp agbcc_arm $1/tools/agbcc/bin/
-		cp -R libc/include $1/tools/agbcc/ #drop include, because we don't want include/include
-		cp ginclude/* $1/tools/agbcc/include/
-		cp libgcc.a $1/tools/agbcc/lib/
-		cp libc.a $1/tools/agbcc/lib/
-		echo "agbcc successfully installed!"
+
+if [ "$1" = "-h" ]; then
+	echo "Usage: install.sh PATH"
+	exit 0
+fi
+
+# If there's no installation directory specified, the install script will default to 
+# installing agbcc into /opt/pret/agbcc
+if [ "$1" = "" ]; then
+	BASE_INSTALL_DIRECTORY=/opt/pret
+	INSTALL_DIRECTORY=$BASE_INSTALL_DIRECTORY
+	mkdir -p $INSTALL_DIRECTORY
+else
+	BASE_INSTALL_DIRECTORY=$1
+	INSTALL_DIRECTORY=$BASE_INSTALL_DIRECTORY/tools
+fi
+
+# The BASE_INSTALL_DIRECTORY nomenclature is so that the check for the existence of the 
+# directory doesn't fail just because there's no tools directory yet.
+if [ -d "$BASE_INSTALL_DIRECTORY" ]; then
```

Comment:
This will fail when `/opt/pret` doesn't exist, forcing people to `mkdir /opt/pret` before running the script.
This is a bit of a significant caveat, and I'd rather the script just create necessary directories itself.

Reply:
If it's imperative to check that a target directory exists, add a `-p` flag or similar to enable implicit directory creation.
Personally, I think it's good form for an installer to create directories automatically, and at worst provide a helpful message when the user doesn't have permissions to create a directory.